src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -1468,6 +1468,15 @@ rpmostree_compose_builtin_extensions (int             argc,
   if (!repo)
     return FALSE;
 
+  rust::String treeref;
+  if (!opt_extensions_base_rev)
+    {
+      treeref = treefile->get_treeref();
+      if (strlen (treeref.c_str()) == 0)
+        return glnx_throw (error, "--base-rev not specified and treefile doesn't have a ref");
+      opt_extensions_base_rev = (char*)treeref.c_str();
```

Comment:
Tricky lifetimes/allocations here.  I feel like it might be cleaner to move the `rust::String` inside the conditional and have this use `g_strdup(treeref.c_str())`.  That way we stick by the rule that our GOption values are C strings backed by `malloc()` - perhaps at some point we actually try to make ASAN/valgrind happy and `free()` those before exiting.